cardstack/cardstack
# New cards show a loading animation while they save 1388

The goal of this PR is to show a "loading" screen in the time between clicking the button to create a blank card and being brought to the schema or edit view. It is implemented as a component used in `application.hbs` and a service because its animation behavior should not be cut off during route transitions. There also should not be multiple active overlays at any given time.

## Added
- An `overlays` service that controls which overlays/modals are visible.
- A `loading` component, referenced from `application.hbs`
- fade animation for the loading component
- tests

## Changed
- Some small testing fixes
- unrelated regression in the context menu positioning (misaligned in view mode)

## Needs feedback

## How to test

To see what this animation will look like in practice, wrap the `save` and transition in a `setTimeout` in `card-name-dialog.js`. Saving to GitHub takes about 3 seconds.

```js
    setTimeout(() => {
      this.overlays.reset();
      if (adoptedFrom) {
        this.router.transitionTo('cards.card.edit.fields', newCard);
      } else {
        this.router.transitionTo('cards.card.edit.fields.schema', newCard);
      }
    }, 3000);
```

The mockup I was working from:

<img width="1056" alt="Screen Shot 2020-03-04 at 2 56 48 PM" src="https://user-images.githubusercontent.com/16627268/75929368-727d0280-5e3e-11ea-800e-e79eac5f6697.png">
